src/widgets/canvas/circle.rs

```diff
@@ -18,12 +20,44 @@ pub struct Circle {
 
 impl Shape for Circle {
     fn draw(&self, painter: &mut Painter<'_, '_>) {
-        for angle in 0..360 {
-            let radians = f64::from(angle).to_radians();
```

Comment:
I like the idea of increasing the resolution for larger circles. Can you elaborate on why something like this wouldn't work?

```suggestion
        let scaling_factor = 100
        for angle in 0..360*scaling_factor {
            let radians = (f64::from(angle) / scaling_factor as f64).to_radians()
```

Reply:
How do you suggest to come up with an ideal scaling factor? Even with a scaling factor that equals the number of pixels we need to draw, it is still possible that the circle line just intersects with one of the pixels at the corner, which can still be skipped.

Reply:
fwiw I haven’t thought about this problem deeply, was just curious why the scaling factor approach wouldn’t work. The only solution I have to your question is that we can make it user configurable.

Reply:
yeah, well I just don't know what value to put there and leaving it to the user sounds weird